openmetadata-service/src/main/java/org/openmetadata/service/migration/api/MigrationWorkflow.java

```diff
@@ -20,16 +21,17 @@ public class MigrationWorkflow {
   private final boolean forceMigrations;
 
   public MigrationWorkflow(
-      Jdbi jdbi, String nativeSQLScriptRootPath, ConnectionType connectionType, boolean forceMigrations) {
+      Jdbi jdbi,
+      String nativeSQLScriptRootPath,
+      ConnectionType connectionType,
+      String extensionSQLScriptRootPath,
+      boolean forceMigrations) {
     this.jdbi = jdbi;
     this.migrationDAO = jdbi.onDemand(MigrationDAO.class);
     this.forceMigrations = forceMigrations;
     // Sort Migration on the basis of version
     List<MigrationFile> availableMigrations =
-        Arrays.stream(Objects.requireNonNull(new File(nativeSQLScriptRootPath).listFiles(File::isDirectory)))
```

Comment:
we should add a unit test around this.
To test different combinations. This one is critical piece of the migrations and any issue there will blow up our migrations

Reply:
intent of test is to use
1. OM bootstrap files does it do ordering correct
2. OM + Collate bootstrap files with OM-1.2.0 , COLLATE-1.2.0, COLLATE-1.2.1 another combo OM-1.2.0, OM-1.2.1 , COLLATE-1.2.2 etc.. different combinations to make sure our ordering is apply OM first, if there is equivalent COLLATE apply after. If there is no OM version but there is COLLATE version we should apply that
3. For each migration we store version that applied we need to make sure Collate version are stored as well

Reply:
added test & screenshot. The result will be as
```
"1.1.0", "1.1.0-collate", "1.2.0", "1.2.1", "1.2.2-collate"
```